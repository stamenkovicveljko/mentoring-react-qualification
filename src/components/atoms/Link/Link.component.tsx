import React from "react";
import * as Styled from "./Link.styles";

interface Props {
  to: string;
  children?: React.ReactNode;
}

const Link = ({ to, children }: Props) => (
  <Styled.Link href={to}>{children || to}</Styled.Link>
);

export default Link;
