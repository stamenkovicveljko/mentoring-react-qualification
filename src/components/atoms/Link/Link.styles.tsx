import styled from "styled-components";
import { linkBlue } from "../../../styles/colors";


export const Link = styled.a`
    color: ${linkBlue};
    font-weight: 600;
`