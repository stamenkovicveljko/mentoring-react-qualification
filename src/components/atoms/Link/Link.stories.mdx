import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import Link from "./Link.component";

<Meta title="Link" component={Link} />

# Link Component
Basic `<a>` element implementation that can work for both inner and outer routes/links.

## Implementation

<Preview>
  <Story name="default">
    <Link to="https://google.com">Google</Link>
  </Story>
</Preview> 

***Note:***

You can use link without specifying child.

<Preview>
  <Story name="without child">
    <Link to="https://google.com"/>
  </Story>
</Preview>

## Props
<Props of={Link} />
