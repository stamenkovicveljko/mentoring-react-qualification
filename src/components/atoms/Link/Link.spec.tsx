import React from "react";
import renderer from "react-test-renderer";
import Link from "./Link.component";

test("Link renders as expected", () => {
  const component = renderer.create(
    <Link to="https://google.com">sample link</Link>
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
