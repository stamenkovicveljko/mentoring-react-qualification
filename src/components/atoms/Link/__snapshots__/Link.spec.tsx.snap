// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders as expected 1`] = `
<a
  className="sc-AxjAm gvFFYS"
  href="https://google.com"
>
  sample link
</a>
`;
